gt/main.c

```diff
@@ -190,6 +238,42 @@ lookup_policy_decision(struct gt_packet_headers *pkt_info,
 	return 0;
 }
 
+static int
+lookup_frag_punish_policy_decision(struct gt_packet_headers *pkt_info,
+	struct ggu_policy *policy, struct gt_instance *instance)
+{
+	policy->flow.proto = pkt_info->inner_ip_ver;
+	if (pkt_info->inner_ip_ver == ETHER_TYPE_IPv4) {
+		struct ipv4_hdr *ip4_hdr =
+			(struct ipv4_hdr *)pkt_info->inner_l3_hdr;
```

Comment:
Given that `pkt_info->inner_l3_hdr` is `void *`, I think you can drop the cast.

The same goes for the IPv6 below.